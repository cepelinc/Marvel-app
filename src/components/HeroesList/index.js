import React from 'react';
import PropTypes from 'prop-types';
import style from './style.styled';
import HeroesListItem from '@components/HeroesListItem';
import PrevNext from '@components/PrevNext';
const Container = style.container;

const HeroesList = ({superHeroes, handleHero, handlePage, offset }) => (
  <Container>
    <h1>MARVEL SUPERHEROES</h1>
    <PrevNext handlePage={handlePage} offset={offset} />
    {superHeroes.map(hero => (
      < HeroesListItem
        name={hero.name}
        key={`id_${hero.id}`}
        id={hero.id}
        comics={hero.comics.available}
        series={hero.series.available}
        stories={hero.stories.available}
        events={hero.events.available}
        picture={hero.thumbnail}
        handleHero={(id) => handleHero(id)}
        offset={offset}
      />
    ))}
  </Container>
)

HeroesList.propTypes = {
  superHeroes: PropTypes.array,
  handleHero: PropTypes.func,
  handlePage: PropTypes.func,
  offset: PropTypes.number
};

export default HeroesList
