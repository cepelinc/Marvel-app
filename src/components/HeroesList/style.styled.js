
import styled from 'styled-components';

export default {
  container: styled.div`
    flex: .35;
    box-shadow: 10px -1px 16px -1px rgba(204,200,204,1);
    height: 100%;
    overflow-y: scroll;
    img {
      width: 50%
    };
    h1{
      margin: 0;
      padding: 1em;
      text-align: center;
      background: blue;
      color: white;
    }
  `
}
