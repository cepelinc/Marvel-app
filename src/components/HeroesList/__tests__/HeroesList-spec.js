import React from 'react'
import Jest from 'jest';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import HeroesList from '@components/HeroesList'

describe('HeroesList', () => {
  const mockData = {
    superHeroes : [{name: 'Spiderman', id: 1, comics: {available: 5}, series: {available: 5}, events: {available: 5}, stories: {available: 5}, thumbnail: {path: 'path', extension: 'jpg'}}],
    handleHero : Jest.func,
    handlePage : Jest.func,
    offset : 1
  }
  it('matches snapshot', () => {
    const Component = renderer.create(
      <BrowserRouter>
        <HeroesList {...mockData} />
      </BrowserRouter>)
      .toJSON()

    expect(Component).toMatchSnapshot()
  })
})
