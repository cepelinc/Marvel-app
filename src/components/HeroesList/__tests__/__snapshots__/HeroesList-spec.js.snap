// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroesList matches snapshot 1`] = `
<div
  className="stylestyled__container-sc-1ibe3y5-0 gExrlQ"
>
  <h1>
    MARVEL SUPERHEROES
  </h1>
  <div
    className="stylestyled__container-sc-1w3xa03-0 bwkBWD"
  >
    <div>
      <span
        onClick={[Function]}
      >
        &lt; prev page
      </span>
    </div>
    <div>
      <span
        onClick={[Function]}
      >
        next page &gt;
      </span>
    </div>
  </div>
  <div
    className="stylestyled__container-u3r3w8-0 bsOeVg"
  >
    <div>
      <div>
        <img
          src="path.jpg"
        />
        <span>
          Spiderman
        </span>
      </div>
      <div>
        <span>
          Comics
        </span>
        <p>
          yeah
        </p>
      </div>
      <div>
        <span>
          Series
        </span>
        <p>
          yeah
        </p>
      </div>
      <div>
        <span>
          Events
        </span>
        <p>
          yeah
        </p>
      </div>
      <div>
        <span>
          Stories
        </span>
        <p>
          yeah
        </p>
      </div>
    </div>
    <a
      className="stylestyled__link-u3r3w8-1 gMsAOd"
      href="/heroes/1"
      onClick={[Function]}
    >
      Find out more about 
      Spiderman
       &gt; &gt; &gt;
    </a>
  </div>
</div>
`;
